Compare pointer to char (build failed gcc 7.3.0)

On ubuntu:bionic with gcc 7.3.0, it complains about:
```
src/ol_app_info.c:463:21: error: comparison between pinter and zero character constant [-Werror=pointer-compare]
       if (*pathiter != '\0')
                     ^~
src/ol_app_info.c:463:11: note: did you mean to dereference the pointer?
       if (*pathiter != '\0')
           ^
```

--- a/src/ol_app_info.c
+++ b/src/ol_app_info.c
@@ -460,7 +460,7 @@ _find_binfile (const gchar *binfile, gboolean match_prefix)
     gchar **pathiter;
     for (pathiter = pathv; *pathiter != NULL; pathiter++)
     {
-      if (*pathiter != '\0')
+      if (**pathiter != '\0')
         path_list = g_list_prepend (path_list, g_strdup (*pathiter));
     }
     path_list = g_list_reverse (path_list);

